# HG changeset patch
# User Rik <rik@octave.org>
# Date 1414529132 25200
#      Tue Oct 28 13:45:32 2014 -0700
# Node ID 64f034147e9ac330e7caa1a8bbab365b9031ec1f
# Parent  93a33123fcfe740611227ad7d8f1698df2dab1bf
test.m: Eliminate needless use of feof().

* test.m (__extract_test_code): Use fgets, which returns a numeric value when
it encounters EOF, to simplify loop extracting test code from m-file.

diff --git a/scripts/testfun/test.m b/scripts/testfun/test.m
--- a/scripts/testfun/test.m
+++ b/scripts/testfun/test.m
@@ -737,23 +737,19 @@ function str = trimleft (str)
   idx = find (! isspace (str), 1);
   str = str(idx:end);
 endfunction
 
 function body = __extract_test_code (nm)
   fid = fopen (nm, "rt");
   body = [];
   if (fid >= 0)
-    while (! feof (fid))
-      ln = fgetl (fid);
-      if (length (ln) >= 2 && strcmp (ln(1:2), "%!"))
-        body = [body, "\n"];
-        if (length (ln) > 2)
-          body = [body, ln(3:end)];
-        endif
+    while (ischar (ln = fgets (fid)))
+      if (strncmp (ln, "%!", 2))
+        body = [body, ln(3:end)];
       endif
     endwhile
     fclose (fid);
   endif
 endfunction
 
 
 ## example from toeplitz
